File: Library/LIbrary/src/library/Books.java
package library;
import java.util.Random;
public class Books {
 private String title;
 private boolean isRented;
 private Random r = new Random();
 private String ID;
  private int rentCounter;
  private Users owner;
    public Books(String title) {
        this.title = title;
        this.isRented = false;
        ID = generateBookID();
    }
    public void setRented(boolean isRented){
        this.isRented=isRented;
    }
    public String getTitle() {
        return title;
    }

    public boolean isRented() {
        return isRented;
    }
    public String getID(){
        return ID;
    }
     public int getRentCounter(){
        return rentCounter;
    }
   
    public void increaseCounter(){
        rentCounter++;
    }

    public void setOwner(Users owner){
        this.owner=owner;
    }

    public Users getOwner(){
        return this.owner;
    }

   public void rentBook(Users owner){
        setOwner(owner);
        setRented(true);
        rentCounter++;
        System.out.println("El libro ha sido rentado correctamente");
    }

    public void returnBook(){
        setOwner(null);
        setRented(false);
    }
    private String generateBookID(){
        String car = "abcdefghijklmnopqrstuvwxyz0123456789";
        String com = "";
        for (int i = 0; i < 6; i++) {
            com += car.charAt(r.nextInt(car.length()));
        }

        return com;
    }

   
}

File: Books.py
import random

class Books:
    def __init__(self, title):
        self.title = title
        self.isRented = False
        self.r = random.Random()
        self.ID = self.generateBookID()
        self.rentCounter = 0
        self.owner = None

    def setRented(self, isRented):
        self.isRented = isRented

    def getTitle(self):
        return self.title

    def isRented(self):
        return self.isRented

    def getID(self):
        return self.ID

    def getRentCounter(self):
        return self.rentCounter

    def increaseCounter(self):
        self.rentCounter += 1

    def setOwner(self, owner):
        self.owner = owner

    def getOwner(self):
        return self.owner

    def rentBook(self, owner):
        self.setOwner(owner)
        self.setRented(True)
        self.rentCounter += 1
        print("El libro ha sido rentado correctamente")

    def returnBook(self):
        self.setOwner(None)
        self.setRented(False)

    def generateBookID(self):
        car = "abcdefghijklmnopqrstuvwxyz0123456789"
        com = ""
        for i in range(6):
            com += car[self.r.randint(0, len(car)-1)]
        return com

